### ISSUE TITLE
Should functions need their own directory?

### ISSUE THREAD

**Issue by joostfarla** [serverless/serverless#429]
The module directory contains all kinds of subdirectories with mixed purposes: one dir per function, lib, package. I think it would be more clean to create a separate directory level for the functions. This would also prevent possible naming conflicts.

Proposed layout:

```
modules
    module1
        lib
        functions
            function1
                event.json
                handler.js
                s-function.json
        s-module.json
```

**Comment by ac360** [comment]
We are thinking about this too and waiting for the right time to introduce this change. :+1:

**Comment by visualasparagus** [comment]
Related to #396

**Comment by joostfarla** [comment]
Fixed in `v0.1.0`.
